presto-main/src/main/java/com/facebook/presto/util/DateTimeUtils.java

```diff
@@ -117,14 +124,26 @@ public static String printDate(int days)
                 .withOffsetParsed();
     }
 
+    public static long parseTimestampLiteral(String value)
+    {
+        try {
+            DateTime dateTime = TIMESTAMP_WITH_TIME_ZONE_FORMATTER.parseDateTime(value);
+            return packDateTimeWithZone(dateTime);
+        }
+        catch (Exception e) {
+            return TIMESTAMP_WITHOUT_TIME_ZONE_FORMATTER.parseMillis(value);
+        }
+    }
+
+    @Deprecated
     public static long parseTimestampLiteral(TimeZoneKey timeZoneKey, String value)
     {
         try {
             DateTime dateTime = TIMESTAMP_WITH_TIME_ZONE_FORMATTER.parseDateTime(value);
             return packDateTimeWithZone(dateTime);
         }
         catch (Exception e) {
-            return TIMESTAMP_WITHOUT_TIME_ZONE_FORMATTER.withChronology(getChronology(timeZoneKey)).parseMillis(value);
+            return LEGACY_TIMESTAMP_WITHOUT_TIME_ZONE_FORMATTER.withChronology(getChronology(timeZoneKey)).parseMillis(value);
```

Comment:
Should not be this dependent to `legacy_timestamp` property to be turned on or to be turned off?

Reply:
It is used only in places where legacy_timestamp is enabled.